.github/workflows/tests.yml

```diff
@@ -8,14 +8,14 @@ jobs:
   tests:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v3.0.2
       - name: Install Python 3
-        uses: actions/setup-python@v3
+        uses: actions/setup-python@v4.1.0
         with:
           python-version: 3.9
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install -r requirements-dev.txt
-      - name: Run tests with pytest
-        run: pytest tests/unit_tests --durations=0
+          pip install tox tox-gh-actions
+      - name: Test with tox
+        run: tox
```

Comment:
Thanks for doing this. We're likely to use tox for more stuff.

Note to team, we need to verify test failures propagate appropriately (locally and in CI) before merging this.